src/index.d.ts

```diff
@@ -545,165 +1005,885 @@ declare namespace admin.messaging {
+
+    /**
+     * Sends all the messages in the given array via Firebase Cloud Messaging.
+     * Employs batching to send the entire list as a single RPC call. Compared
+     * to the `send()` method, this method is a significantly more efficient way
+     * to send multiple messages.
+     *
+     * The responses list obtained from the return value
+     * corresponds to the order of tokens in the `MulticastMessage`. An error
+     * from this method indicates a total failure -- i.e. none of the messages in
+     * the list could be sent. Partial failures are indicated by a `BatchResponse`
+     * return value.
+     *
+     * @param {!Array<!admin.messaging.Message>} messages A non-empty array
+     *   containing up to 100 messages.
+     * @param {boolean=} dryRun Whether to send the messages in the dry-run
+     *   (validation only) mode.
+     * @return {!Promise<!admin.messaging.BatchResponse>} A Promise fulfilled
+     *   with an object representing the result of the send operation.
+     */
     sendAll(
       messages: Array<admin.messaging.Message>,
       dryRun?: boolean
     ): Promise<admin.messaging.BatchResponse>;
+
+    /**
+     * Sends the given multicast message to all the FCM registration tokens
+     * specified in it.
+     *
+     * This method uses the `sendAll()` API under the hood to send the given
+     * message to all the target recipients. The responses list obtained from the
+     * return value corresponds to the order of tokens in the `MulticastMessage`.
+     * An error from this method indicates a total failure -- i.e. the message was
+     * not sent to any of the tokens in the list. Partial failures are indicated by
+     * a `BatchResponse` return value.
+     *
+     * @param {!admin.messaging.MulticastMessage} message A multicast message
+     *   containing up to 100 tokens.
+     * @param {boolean=} dryRun Whether to send the message in the dry-run
+     *   (validation only) mode.
+     * @return {!Promise<!admin.messaging.BatchResponse>} A Promise fulfilled
+     *   with an object representing the result of the send operation.
+     */
     sendMulticast(
       message: admin.messaging.MulticastMessage,
       dryRun?: boolean
     ): Promise<admin.messaging.BatchResponse>;
+
+    /**
+     * Sends an FCM message to a single device corresponding to the provided
+     * registration token.
+     *
+     * See
+     * [Send to individual devices](/docs/cloud-messaging/admin/legacy-fcm#send_to_individual_devices)
+     * for code samples and detailed documentation. Takes either a 
+     * `registrationToken` to send to a single device or a
+     * `registrationTokens` parameter containing an array of tokens to send
+     * to multiple devices.
+     *
+     * @param {string} registrationToken The registration token for the device to
+     *   which to send the message.
+     * @param {!Array<string>} registrationTokens An array of registration tokens
```

Comment:
This doesn't look correct. I think this description should be merged with the description of the `registrationToken` field. Something like:

```
@param registrationToken A device registration token or an array of
  device registration tokens to which the message should be sent.
```